### PR TITLE
Java sol for usaco 645

### PR DESCRIPTION
**Pull request by Keys9873**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

**Comment by Keys9873** [comment]
I translated the C++ code into Java.

File: solutions/silver/usaco-645.mdx [hunk 1]
```diff
@@ -112,6 +112,99 @@ int main() {
...
+		first = x;
+		second = y;
+	}
+	public int compareTo(Pair o) {
+		if (o.first != first) return (int)(first - o.first);
+		return (int)(second - o.second);
+	}
+}
+
+public class main {
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-645.mdx:132]:
```suggestion
public class Main {
```

File: solutions/silver/usaco-645.mdx [hunk 2]
```diff
@@ -112,6 +112,99 @@ int main() {
...
+		br.close();
+
+		long ans = 0;
+
+		// finds the maximum area saved by testing splits along cows[i].first
+		for (int t = 0; t < 2; t++) {
+			Arrays.sort(cows);
+			for (int i = 0; i < n; i++) {
+				System.out.println(cows[i].first + " " + cows[i].second);
+			}
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-645.mdx:159]: uh

**Review by ryanchou-dev — COMMENTED** [review]
woohoo

File: solutions/silver/usaco-645.mdx [hunk 3]
```diff
@@ -112,6 +112,96 @@ int main() {
 }
 ```
 </CPPSection>
+<JavaSection>
+```java
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-645.mdx:116]:
```suggestion
<JavaSection>

```java
```

**Review by ryanchou-dev — COMMENTED** [review]
nice!

im not too good at reviewing java code loll, @SansPapyrus683 ?

File: solutions/silver/usaco-645.mdx [hunk 4]
```diff
@@ -112,6 +112,97 @@ int main() {
 }
 ```
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+class Pair implements Comparable<Pair> {
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-645.mdx:121]: put @ bottom and wrap in codesnip

File: solutions/silver/usaco-645.mdx [hunk 5]
```diff
@@ -112,6 +112,97 @@ int main() {
...
+		first = x;
+		second = y;
+	}
+	public int compareTo(Pair o) {
+		if (o.first != first) return (int)(first - o.first);
+		return (int)(second - o.second);
+	}
+}
+
+public class Main {
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-645.mdx:133]: use problem name instead

File: solutions/silver/usaco-645.mdx [hunk 6]
```diff
@@ -112,6 +112,97 @@ int main() {
...
+
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-645.mdx:134]:
```suggestion
\```

File: solutions/silver/usaco-645.mdx [hunk 7]
```diff
@@ -112,6 +112,97 @@ int main() {
...
+	// returns updated min/maxs when considering a new coordinate
+	public static Pair upd(Pair a, long b) {
+		return new Pair(Math.min(a.first, b), Math.max(a.second, b));
+	}
+
+	public static void main(String[] args) throws Exception {
+		BufferedReader br = new BufferedReader(new FileReader("split.in"));
+		StringTokenizer st;
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-645.mdx:142]: declare this in the for loop

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/usaco-645.mdx (hunk 7 above)
Comment (Keys9873) [reply solutions/silver/usaco-645.mdx:143]: Why? Isn't it better to declare one and initialize several times?

File: solutions/silver/usaco-645.mdx [hunk 8]
```diff
@@ -112,6 +113,100 @@ int main() {
...
+import java.io.*;
+import java.util.*;
+
+public class SplittingTheField {
+	\```
+	    // returns updated min/maxs when considering a new coordinate
+	    public static Pair
+	    upd(Pair a, long b) {
+		return new Pair(Math.min(a.first, b), Math.max(a.second, b));
+	}
```
Comment (ryanchou-dev) [inline solutions/silver/usaco-645.mdx:128]: wait is the [ \\``` ] a mistake or am i high

**Review by ryanchou-dev — COMMENTED** [review]
ily keys

File: solutions/silver/usaco-645.mdx (hunk 8 above)
Comment (Keys9873) [reply solutions/silver/usaco-645.mdx:128]: it was there when i found it, i got rid of it -> build failed -> i put it back -> build failed again

**Review by SansPapyrus683 — APPROVED** [review]
honestly would prefer you make the code consistent w/ the other solutions but whatever

File: solutions/silver/usaco-645.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/silver/usaco-645.mdx]: is it? can i have a so link if that's the case

File: solutions/silver/usaco-645.mdx [hunk 9]
```diff
@@ -6,32 +6,6 @@ title: Splitting the Field
 author: Óscar Garries, Benjamin Qi, Ryan Chou
 ---
 
-[Official Analysis (Java)](http://www.usaco.org/current/data/sol_split_gold_open16.html)
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-645.mdx:8]: ayo what

Comment (Keys9873) [reply solutions/silver/usaco-645.mdx:8]: Im not sure what is causing the failure I am just removing parts to find what it is.

**Comment by SansPapyrus683** [comment]
how about you just remake the pr entirely

**Pull request closed without merging**